### PR TITLE
cpu: aarch64: batch_normalization : Expand ARM SVE support in jit_uni_batch_normalization

### PR DESCRIPTION
**Pull request by nikhilfujitsu**
## Description
This commit enhances the existing ARM SVE support in jit_uni_batch_normalization to include additional vector length compatibility. The changes made are for implementation of different ARM SVE vector length.

## Major Code changes:

Updated the block size definition to accommodate other SVE vector length.
Added 'OR' and 'AND' conditions to extend support for other SVE vector.
Predicate registers are set according to isa vector length.
ldr and str instructions are replaced by ld1w and st1w respectively. For appropriate load and strore operation considering isa vector length.

## Checklist
General
[✓] Do all unit and benchdnn tests (make test and make test_benchdnn_*) pass locally for each commit? Yes
Test output is same with and without this commit.

make test output:
99% tests passed, 1 tests failed out of 103

Total Test time (real) = 36.37 sec

The following tests FAILED:
102 - test_graph_unit_cpu (Failed)
Errors while running CTest
Output from these tests are in: /home/nikhil/TEST/oneDNN/build/Testing/Temporary/LastTest.log
Use "--rerun-failed --output-on-failure" to re-run the failed cases verbosely.
make: *** [Makefile:71: test] Error 8

make test_benchdnn_* output:
make: *** No rule to make target 'test_benchdnn_*'. Stop.

make test_benchdnn_bnorm_ci_cpu/fast output.
tests:4445 passed:1232 skipped:2989 mistrusted:224 unimplemented:0 invalid_arguments:0 failed:0 listed:0

### CODE REVIEW

**Review by mgouicem — APPROVED** [review]
Thank you for the contribution. Tagging @jondea

**Review by jondea — COMMENTED** [review]
Thank you for the contribution, we really appreciate your work on vlen 256. I just have a few comments about whether it could be more general. If it works for an SVE vlen of 256 and 512, why not any length? Then you could remove any hard coded vlen variables, and there could be a single implementation.

File: src/cpu/aarch64/jit_uni_batch_normalization.cpp [hunk 1]
```diff
@@ -1080,13 +1087,13 @@ struct jit_bnorm_t : public jit_generator {
                     subs_imm(reg_ctr, reg_ctr, 1, X_TMP_0);
                     b(NE, mean_reduction_thrs);
                 }
-                if (isa == sve_512)
+                if (isa == sve_256 || isa == sve_512)
```
[the remark this answers is not in the document]
Comment (nikhilfujitsu) [reply src/cpu/aarch64/jit_uni_batch_normalization.cpp:1090]: Thanks for the remarks @jondea . We are happy to incorporate suggested changes in future work.

File: src/cpu/aarch64/jit_uni_batch_normalization.cpp [hunk 2]
```diff
@@ -1004,7 +1010,8 @@ struct jit_bnorm_t : public jit_generator {
         L(zero_rbuf);
         {
             uni_str(TReg(0), reg_rbuf1, reg_coff);
-            add_imm(reg_coff, reg_coff, isa == sve_512 ? vlen : vlen / 2,
+            add_imm(reg_coff, reg_coff,
+                    (isa == sve_256 || isa == sve_512) ? vlen : vlen / 2,
```
[the remark this answers is not in the document]
Comment (nikhilfujitsu) [reply src/cpu/aarch64/jit_uni_batch_normalization.cpp:1014]: Indeed @jondea . We will incorporate suggested updates in the future PR. Thanks for the remarks.

**Review by jondea — APPROVED** [review]

**Comment by abhijain1204fujitsu** [comment]
@vpirogov , Hello
Can you please merge the PR as the changes are approved
In case anything is required from our end, do let us know.

**Pull request merged**